Storage/Processor/Processor.php

```diff
@@ -207,26 +207,39 @@ private function generateImage(Config $config, string $filename = null): string
         return $generatedImage;
     }
 
+    private function getResourceFromAsset(Config $config)
+    {
+        if ($config->getFilename()) {
+            return fopen($config->getFilename(), 'r');
+        }
 
-    public function getResource(Config $config, string $filename)
+        return $this->storageManager->getResource($config->getAssetHash(), $config->getStorageContext());
+    }
+
+    public function getResource(Config $config, string $filename, bool $noCache = false)
     {
-        $cacheableResult = $config->cacheableResult();
         if (!$config->getStorageContext()) {
-            if ($config->getFilename()) {
-                return fopen($config->getFilename(), 'r');
-            }
+            return $this->getResourceFromAsset($config);
+        }
 
-            return $this->storageManager->getResource($config->getAssetHash(), $config->getStorageContext());
+        if ($config->isCacheableResult() && !$noCache) {
+            try {
+                return $this->storageManager->getResource($config->getAssetHash(), $config->getConfigHash());
+            } catch (NotFoundException $e) {
+            } catch (\Exception $e) {
+                $this->logger->warning('log.unexpected_exception', ['error_message' => $e->getMessage()]);
+            }
         }
 
-        try {
-            return $this->storageManager->getResource($config->getAssetHash(), $config->getConfigHash());
-        } catch (NotFoundException $e) {
-            $generatedResource = $this->generateResource($config);
-            if ($cacheableResult) {
+        $generatedResource = $this->generateResource($config);
```

Comment:
This should return and generateResource should try to cache the generated result?

Reply:
I though about it but the generateRessource function would become less clear and longer. And as this function is not too long it was better like this to me.

Reply:
But here it adds to complexity with all the if, try, ... or trying to cache should be a function on itself that can be used (if other parts also try to cache?)

Reply:
it has been refactored with getResourceFromCache and saveResourceToCache functions without nested try/if